lib/index.js

```diff
@@ -80,6 +86,13 @@ Optimizely.prototype.track = function(track) {
   var revenue = track.revenue();
   if (revenue) payload.revenue = revenue *= 100;
   push(['trackEvent', track.event(), payload]);
+
+  var optimizelyClientInstance = window.optimizelyClientInstance;
+  if (optimizelyClientInstance && optimizelyClientInstance.track) {
+    var attributes = track.properties();
+    var userId = attributes.userId;
```

Comment:
@hankim813 I am not testing it on the gist. I have a node web server running that serves that HTML page. I tried what you suggested but I still get the same result, where `track.userId()` is undefined even after calling `analytics.identify('user~1')`. 

In any case, I've updated the code and added `track.userId()` as a default if `userId` is not passed in as a property. Doing it this way because we still want the user to specify the visitor id if the know the user is not logged in. However, if the user is logged in then they can rely on `track.userId`.